qiskit/backends/ibmq/ibmqjob.py

```diff
@@ -310,126 +291,122 @@ def validating(self):
             try:
                 # Use CamelCase when passing the hpc parameters to the API.
-                hpc = {
-                    'multiShotOptimization':
-                        qobj['config']['hpc']['multi_shot_optimization'],
-                    'ompNumThreads':
-                        qobj['config']['hpc']['omp_num_threads']
+                hpc_camel_cased = {
+                    'multiShotOptimization': hpc['multi_shot_optimization'],
+                    'ompNumThreads': hpc['omp_num_threads']
                 }
             except (KeyError, TypeError):
-                hpc = None
-        backend_name = qobj['config']['backend_name']
-        if backend_name != self._backend_name:
-            raise QISKitError("inconsistent qobj backend "
-                              "name ({0} != {1})".format(backend_name,
-                                                         self._backend_name))
+                hpc_camel_cased = None
+
+        self._future = self._executor.submit(self._submit_callback, api_jobs,
+                                             self._backend_name, hpc_camel_cased,
+                                             seed, shots, max_credits)
+
+    def _submit_callback(self, api_jobs, backend_name, hpc, seed, shots, max_credits):
+        """ Submit job to IBM Q.
+
+        Args:
+            api_jobs (list): List of API Job dictionaries to submit. One per circuit.
+            backend_name (string): The name of the backend
+            hpc (dict): HPC specific configuration
+            seed (integer): The seed for the circuits
+            shots (integer): Number of shots the circuits should run
+            max_credits (integer): Maximum number of credits
+
+        Returns:
+            dict: A dictionary with the response of the submitted job
+        """
         try:
             submit_info = self._api.run_job(api_jobs, backend=backend_name,
-                                            shots=qobj['config']['shots'],
-                                            max_credits=qobj['config']['max_credits'],
-                                            seed=seed0,
-                                            hpc=hpc)
-        # pylint: disable=broad-except
-        except Exception as err:
-            self._status = JobStatus.ERROR
-            self._status_msg = str(err)
-            self._exception = err
-            return None
+                                            shots=shots, max_credits=max_credits,
+                                            seed=seed, hpc=hpc)
+        except Exception as err:  # pylint: disable=broad-except
+            # This is not a controlled error form the API, so we don't want to change
+            # job status to ERROR, because we really don't know what happened with the
+            # Job at this point... it could be successfully QUEUED for example.
+            self._future_exception = err
+            return {'error': str(err)}
+
+        # Fail fast!!
```

Comment:
I hate to be the killjoy but remove this comment, please. :P

Reply:
gngngngn